byteview.go

```diff
@@ -158,3 +158,16 @@ func (v ByteView) ReadAt(p []byte, off int64) (n int, err error) {
 	}
 	return
 }
+
+// WriteTo implements io.WriterTo on the bytes in v.
+func (v ByteView) WriteTo(w io.Writer) (int64, error) {
+	b := v.b
+	if b == nil {
+		b = []byte(v.s)
```

Comment:
Ah - I'm still learning what allocates and what doesn't. Thanks for pointing that out.